src/main/scala/com/moesif/filter/MoesifApiFilter.scala

```diff
@@ -36,10 +40,48 @@ class MoesifApiFilter @Inject()(config: MoesifApiFilterConfig)(implicit mat: Mat
   private val moesifCollectorEndpoint = config.moesifCollectorEndpoint
   private val samplingPercentage = config.samplingPercentage
   private val eventModelBuffer = mutable.ArrayBuffer[EventModel]()
+  private val client = new MoesifAPIClient(moesifApplicationId, moesifCollectorEndpoint)
+  private val moesifApi = client.getAPI
+  private var lastTimeAppConfigFetched: DateTime = DateTime.parse("1970-01-01", DateTimeFormat.forPattern("yyyy-MM-dd"))
+  private var isAppConfigFetched: Boolean = false
+  private var appConfigModel = new AppConfigModel
 
   private val logger = Logger.getLogger("moesif.play.filter.MoesifApiFilter")
   logger.info(s"config  is $config")
 
+  def getSampleRateToUse(userId: String, companyId: String): Int = {
+    var sampleRate = appConfigModel.getSampleRate
+    if (userId != null && appConfigModel.getUserSampleRate.containsKey(userId)) sampleRate = appConfigModel.getUserSampleRate.get(userId)
+    else if (companyId != null && appConfigModel.getCompanySampleRate.containsKey(companyId)) sampleRate = appConfigModel.getCompanySampleRate.get(companyId)
+    sampleRate
+  }
+
+  def getApplicationConfig() = {
+
+    if (!isAppConfigFetched || DateTime.now(DateTimeZone.UTC).isAfter(lastTimeAppConfigFetched.plusMinutes(5))) {
```

Comment:
this fcn to be synchronized.

Reply:
actually I think correct way to do this, have a separate thread which runs every 5 mins, and update the appConfigModel as needed. 

Regualar threads shouldn't have to make additional HTTP call in their flow.